Source/WebCore/html/HTMLCanvasElement.cpp

```diff
@@ -453,6 +455,13 @@ WebGLRenderingContextBase* HTMLCanvasElement::createContextWebGL(WebGLVersion ty
     if (!shouldEnableWebGL(document().settings()))
         return nullptr;
 
+#if PLATFORM(MAC)
+    if (!isGPUAvailable()) {
```

Comment:
Why is it relevant for the WebProcess to have Metal access when we do WebGL in the GPUProcess nowadays?

Reply:
WebGL runs in the WebContent process on macOS.

Reply:
Then this should be probably controlled based on some sort of feature flag, not a PLATFORM(MAC) which will bite us later when we do add support on macOS.

Reply:
Like add a WebCore binding for WebKit's UseGPUProcessForWebGLEnabled feature flag and rely on this instead of a `#if PLATFORM(MAC)` compile flag.

Reply:
There's a setting: UseGPUProcessForWebGLEnabled. Let's use that.

Reply:
> Like add a WebCore binding for WebKit's UseGPUProcessForWebGLEnabled feature flag and rely on this instead of a `#if PLATFORM(MAC)` compile flag.

Yes, that is a good point.

Thanks for reviewing!

Reply:
> There's a setting: UseGPUProcessForWebGLEnabled. Let's use that.

Fixed in latest patch.

Thanks for reviewing!